src/libs/ReactNativeLauncher.js

```diff
@@ -269,11 +271,21 @@ class ReactNativeLauncher extends Launcher {
    *
    * @param {SetWorkerStateOptions} options
    */
-  async setWorkerState(options) {
-    this.emit('SET_WORKER_STATE', options)
-    if (options?.visible === true) {
-      this.emit('worker:visible')
-    }
+  setWorkerState(options) {
+    return new Promise((resolve, reject) => {
+      let timerId = null
+      this.emit('SET_WORKER_STATE', options)
+      this.once('worker:webview:ready', () => {
+        clearTimeout(timerId)
+        if (options?.visible === true) {
+          this.emit('worker:visible')
+        }
+        resolve()
+      })
+      timerId = setTimeout(() => {
+        reject('ReactNativeLauncher.setWorkerState took more than 30000 ms')
```

Comment:
nit: Shouldn't we start using const to have some kind of errors constant to match them easily within the app? Matching string is error prone. 

```

reject(ERRORS.SET_WORKER_STATE_TOO_LONG_TO_INIT)

...
catch(e){
if(e.message === ERRORS.SET_WORKER_STATE_TOO_LONG_TO_INIT) {
...
}
}

```

Like that, we can also have a quick views at all the errors we can have.

const ERRORS = [...]